### PR TITLE
Refactor responses to match geocodejson spec

### PR DESCRIPTION
**Pull request by dianashk**
Fixes pelias/api#201

For a request such as 
> http://pelias.mapzen.com/search?input=samsung%20accelerator&size=2

Responses went from this: 

```javascript
{
  "type":"FeatureCollection",
  "features":[
    {
      "type":"Feature",
      "properties":{
        "id":"2688438317",
        "layer":"osmnode",
        "name":"Samsung Accelerator",
        "alpha3":"USA",
        "admin0":"United States",
        "admin1":"New York",
        "admin1_abbr":"NY",
        "admin2":"New York County",
        "local_admin":"Manhattan",
        "locality":"New York",
        "neighborhood":"Flatiron District",
        "category":[

        ],
        "address":{
          "zip":"10010",
          "number":"30",
          "street":"West 26th Street"
        },
        "text":"Samsung Accelerator, Manhattan, NY"
      },
      "geometry":{
        "type":"Point",
        "coordinates":[
          -73.990342,
          40.744243
        ]
      }
    },
    {
      "type":"Feature",
      "properties":{
        "id":"257190553",
        "layer":"osmway",
        "name":"Accelerator",
        "alpha3":"USA",
        "admin0":"United States",
        "admin1":"Idaho",
        "admin1_abbr":"ID",
        "admin2":"Blaine County",
        "locality":"Sun Valley",
        "category":[
          "transport"
        ],
        "address":{

        },
        "text":"Accelerator, Sun Valley, ID"
      },
      "geometry":{
        "type":"Point",
        "coordinates":[
          -114.347904,
          43.68922
        ]
      }
    }
  ],
  "bbox":[
    -114.347904,
    40.744243,
    -73.990342,
    43.68922
  ],
  "date":1441082272781
}
```
to this:

```javascript
{
  "geocoding":{
    "version":"0.1",
    "license":"undefined/license",
    "attribution":"undefined/attribution",
    "query":{
      "input":"samsung accelerator",
      "parsed_input":{
        "target_layer":[
          "geoname",
          "osmnode",
          "osmway",
          "admin0",
          "admin1",
          "admin2",
          "neighborhood",
          "locality",
          "local_admin"
        ]
      },
      "size":2,
      "default_layers_set":true,
      "layers":[
        "geoname",
        "osmnode",
        "osmway",
        "admin0",
        "admin1",
        "admin2",
        "neighborhood",
        "locality",
        "local_admin",
        "osmaddress",
        "openaddresses"
      ],
      "details":true
    },
    "engine":{
      "name":"Pelias",
      "author":"Mapzen"
    },
    "timestamp":1441082453784
  },
  "type":"FeatureCollection",
  "features":[
    {
      "type":"Feature",
      "properties":{
        "geocoding":{
          "id":"2688438317",
          "type":"venue",
          "source":"osm",
          "name":"Samsung Accelerator",
          "housenumber":"30",
          "street":"West 26th Street",
          "category":[

          ],
          "postalcode":"10010",
          "country_a":"USA",
          "country":"United States",
          "region":"New York",
          "region_a":"NY",
          "county":"New York County",
          "localadmin":"Manhattan",
          "locality":"New York",
          "neighbourhood":"Flatiron District",
          "label":"Samsung Accelerator, Manhattan, NY"
        }
      },
      "geometry":{
        "type":"Point",
        "coordinates":[
          -73.990342,
          40.744243
        ]
      }
    },
    {
      "type":"Feature",
      "properties":{
        "geocoding":{
          "id":"54229742",
          "type":"venue",
          "source":"osm",
          "name":"Accelerator",
          "category":[

          ],
          "country_a":"NOR",
          "country":"Norway",
          "region":"Oslo",
          "county":"Bydel Grorud",
          "locality":"Oslo",
          "neighbourhood":"Ammerud",
          "label":"Accelerator, Oslo"
        }
      },
      "geometry":{
        "type":"Point",
        "coordinates":[
          10.852097,
          59.950827
        ]
      }
    }
  ],
  "bbox":[
    -73.990342,
    40.744243,
    10.852097,
    59.950827
  ]
}
```

### CODE REVIEW

**Comment by orangejulius** [comment]
All the refactoring here that splits lots of behavior out into friendly little functions is excellent and much needed!

File: routes/legacy.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
 var proxy = require('express-http-proxy');
 
 function addRoutes(app, peliasConfig) {
+  var sendToLegacy;
+
   if (!peliasConfig.hasOwnProperty('legacyUrl')) {
-    return;
+    sendToLegacy = function doNothing(req, res, next) {
+      next(new Error('Invalid path, no legacy proxy specified'));
```
Comment (orangejulius) [inline routes/legacy.js]: This is probably better than silently disabling the legacy routing :)

File: routes/v1.js [hunk 2]
```diff
@@ -1,36 +1,91 @@
...
 function addRoutes(app, peliasConfig) {
-  // api root
-  app.get( '/v1/', controllers.index() );
 
-  // place API
-  app.get( '/v1/place', sanitisers.place.middleware, controllers.place() );
+  /** ------------------------- routers ------------------------- **/
 
-  // suggest APIs
-  app.get( '/v1/suggest', sanitisers.search.middleware, controllers.search() );
```
Comment (orangejulius) [inline routes/v1.js]: We'll bring this back separately as autocomplete, right?

File: routes/v1.js [hunk 3]
```diff
@@ -1,36 +1,91 @@
...
-  app.get( '/v1/search/coarse', sanitisers.coarse.middleware, controllers.search() );
 
-  // reverse API
-  app.get( '/v1/reverse', sanitisers.reverse.middleware, controllers.search(undefined, require('../query/reverse')) );
+  var base = '/v1/';
+
+  // api root
+  app.get ( base,                  routers.index );
+  app.get ( base + 'place',        routers.place );
+  app.get ( base + 'autocomplete', routers.search );
```
Comment (dianashk) [inline routes/v1.js]: `autocomplete` is here :)

File: routes/v1.js (hunk 2 above)
Comment (dianashk) [inline routes/v1.js]: see below :arrow_down: :)

File: routes/v1.js [hunk 4]
```diff
@@ -1,36 +1,91 @@
...
+      controllers.place(),
+      postProc.renamePlacenames(),
+      postProc.geocodeJSON(peliasConfig),
+      postProc.sendJSON
+    ])
+  };
 
-  // search APIs
-  app.get( '/v1/search', sanitisers.search.middleware, controllers.search() );
-  app.get( '/v1/search/coarse', sanitisers.coarse.middleware, controllers.search() );
```
Comment (orangejulius) [inline routes/v1.js]: So this change also removes the coarse endpoint? Excellent

File: routes/v1.js (hunk 3 above)
Comment (orangejulius) [inline routes/v1.js]: oh, nice

**Pull request merged**